pkg/kube/controllers/extendedjob/job_controller.go

```diff
@@ -54,7 +54,7 @@ func AddJob(ctx context.Context, config *config.Config, mgr manager.Manager) err
 				return false
 			}
 
-			shouldProcessEvent := o.Status.Succeeded == 1 || o.Status.Failed == 1
+			shouldProcessEvent := o.Status.Succeeded == 1 || o.Status.Failed == (*o.Spec.BackoffLimit+1)
```

Comment:
Not sure if this can actually happen, but maybe the `BackoffLimit` could be decreased by the user and thus the `Failed` count becomes greather than `BackoffLimit+1`?

```suggestion
			shouldProcessEvent := o.Status.Succeeded == 1 || o.Status.Failed >= (*o.Spec.BackoffLimit+1)
```